robottelo/virtwho.py

```diff
@@ -0,0 +1,44 @@
+"""Utilities for Virtwho Configure Plugin Testing
+
+All the virtwho-configure UI/CLI/API interfaces should be defined there.
+
+"""
+from robottelo.config import settings
+
+
+def get_hypervisor_type():
+    return{
+        'esx': 'VMware vSphere / vCenter (esx)',
+        'xen': 'XenServer (xen)',
+        'rhevm': 'Red Hat Virtualization Hypervisor (rhevm)',
+        'hyperv': 'Microsoft Hyper-V (hyperv)',
+        'libvirt': 'libvirt',
+        'kubevirt': 'Container-native virtualization',
+    }
+
+
+def get_form_data(
+        name, debug=True, interval='Every hour', hypervisor_id='hostname'):
```

Comment:
My original idea was define all the functions in the virtwho.py, because we have some functions need to be used by ui/api/cli, such as:
 virtwho_logfile_analyzer()
 deploy_by_hammer()
 deploy_by_script()
 register_guest()

An example to understand, the UI have to involve some CLI function:
1. create a esx config by UI
2. get the deploy command line from UI
3. run the command line in virt-who host by CLI
4. check the virtwho log message by CLI
5. check the hypervisor/guest sent to satellite or not by UI
6. attach a physical sku to hypervisor by UI
7. check the virtual sku is generated or not and then attach it to the guest by UI

Reply:
I have to pass name parameter to `get_form_data` to create the UI form, and also need some conditional statements for different hypervisors' form data.  I have no good idea how to pass name parameter to `get_form_data` if I change it to a fixture.

Reply:
>  I have no good idea how to pass name parameter to `get_form_data` if I change it to a fixture.

There are at least three ways. I would consider them more or less the same, so just pick whichever you find easiest to implement and reason about.

1. [Use indirect parametrization](https://stackoverflow.com/a/33879151/3552063)
2. [Create fixture that will return a function](https://stackoverflow.com/a/28570677/3552063) and call that function with parameters in test
3. Let fixture return part of the dict that is constant and update this dict in test. This will be fine here, as you are just taking parameters and putting them in dict, they are not involved in any decision-making.

Reply:
Make sense to me, drop the utils.py now, and update the form_data in test